server.cpp

```diff
@@ -85,23 +146,33 @@ int main(int argc, char* argv[])
         tcp::resolver resolver(io_service);
         tcp::acceptor acceptor(io_service, *resolver.resolve(tcp::resolver::query(host, port)));
 
+        namespace pls = std::placeholders;
+
+        char buff[1024];
+
         for (;;)
         {
             tcp::socket socket(io_service);
             acceptor.accept(socket);
 
-            const std::string message = make_daytime_string();
+            const size_t bytes = read(socket, boost::asio::buffer(buff), std::bind(read_complete, buff, pls::_1, pls::_2));
 
-            boost::system::error_code ignored_error;
-            boost::asio::write(socket, boost::asio::buffer(message), ignored_error);
+            const std::string msg(buff, bytes);
+            const size_t str_end_pos = msg.find("\r");
+            const std::string start_str = msg.substr(0, str_end_pos);
 
-            std::cout << message << " " << socket.remote_endpoint().address().to_string() << "\n";
+            Request request(start_str);
+
+            const std::string date = write_response(socket, request);
+
+            std::string log_message = date + " " + socket.remote_endpoint().address().to_string() + " " + start_str;
```

Comment:
`log_message` is only used once, you can compose the string right in the `write_log` parameter.